execution/aggregate/hashaggregate.go

```diff
@@ -130,13 +134,25 @@ func (a *aggregate) Next(ctx context.Context) ([]model.StepVector, error) {
 	return result, nil
 }
 
-func (a *aggregate) initializeTables(ctx context.Context) error {
+func (a *aggregate) initialize(ctx context.Context) error {
 	var (
 		tables []aggregateTable
 		series []labels.Labels
+		arg    float64
 		err    error
 	)
 
+	if a.paramOp != nil {
+		p, err := a.paramOp.Next(ctx)
```

Comment:
Let's call `a.paramOp.Next` within `Next` instead of `initialize`, similar to how we do it for `a.next`. Here we're calling `Next` even during `Series` calls which might not be ideal. 🙂

Reply:
Humm...

Can `Next` be called 2 times? I'm wondering if i need to make sure `a.paramOp.Next` is only called once.

Reply:
Ok...

I change the PR to receive the `args` on the `aggregate` function:

```
aggregate(arg float64, vector model.StepVector)
```

I think it makes sense as the aggregations has indeed 2 parameters (the arg and the vector)

Reply:
Maybe you can look at benchmarks and see if this affects it in anyway?

Reply:
Seems pretty much the same:

Before: https://gist.github.com/alanprot/bb096702ce42550c75b1aef534bcb7f2
After: https://gist.github.com/alanprot/b6981295d587a81763144cdd5e35c879